modules/core/shared/src/main/scala-3/io/circe/derivation/ConfiguredEncoder.scala

```diff
@@ -22,8 +22,10 @@ trait ConfiguredEncoder[A](using conf: Configuration) extends Encoder.AsObject[A
     val (constructorName, json) = encodeElemAt(index, a, conf.transformConstructorNames)
     conf.discriminator match
       case None => JsonObject.singleton(constructorName, json)
-      case Some(discriminator) =>
-        json.asObject.getOrElse(JsonObject.empty).add(discriminator, Json.fromString(constructorName))
+      case Some(discriminator) => {
+        val jo = json.asObject.getOrElse(JsonObject.empty)
+        if(jo.contains(discriminator)) jo else jo.add(discriminator, Json.fromString(constructorName))
```

Comment:
This is to prevent the Encoder of the mid-level from overriding the discriminator. Although, I am not sure how stable this order of Encoder is. There might be better ways.